src/routes/(app)/schedule/[id]/rsvp/[slug]/+page.server.ts

```diff
@@ -0,0 +1,90 @@
+import { SMTP_USER } from '$env/static/private';
+import { collections } from '$lib/server/database';
+import { queueEmail } from '$lib/server/email';
+import { zodNpub } from '$lib/server/nostr';
+import { runtimeConfig } from '$lib/server/runtime-config';
+import { error } from '@sveltejs/kit';
+import { format } from 'date-fns';
+import { ObjectId } from 'mongodb';
+import { Kind } from 'nostr-tools';
+import { z } from 'zod';
+
+export const load = async ({ params }) => {
+	const schedule = await collections.schedules.findOne({ _id: params.id });
+
+	if (!schedule) {
+		throw error(404, 'schedule not found');
+	}
+	const pictures = await collections.pictures
+		.find({ 'schedule._id': params.id })
+		.sort({ createdAt: 1 })
+		.toArray();
+	return {
+		schedule,
+		pictures
+	};
+};
+
+export const actions = {
+	notify: async function ({ request, params }) {
+		const schedule = await collections.schedules.findOne({ _id: params.id });
+		if (!schedule) {
+			throw error(404, 'Schedule not found');
+		}
+		const eventSchedule = schedule.events.find((e) => e.slug === params.slug);
+		if (!eventSchedule) {
+			throw error(404, 'Event not found');
+		}
+
+		const data = await request.formData();
+
+		const { address } = z
+			.object({
+				address: z.union([z.string().email(), zodNpub()])
+			})
+			.parse({
+				address: data.get('address')
+			});
+		await queueEmail(
+			runtimeConfig.sellerIdentity?.contact.email || SMTP_USER,
+			'schedule.rsvp.admin',
+			{
+				brandName: runtimeConfig.brandName,
+				eventName: eventSchedule.title,
+				participantContact: address
+			}
+		);
+		if (address.includes('@')) {
+			await queueEmail(address || SMTP_USER, 'schedule.rsvp.user', {
+				brandName: runtimeConfig.brandName,
+				eventName: eventSchedule.title,
+				eventDate: format(eventSchedule.beginsAt, 'EEEE dd MMM yyyy HH:mm'),
+				eventShortDescription: eventSchedule.shortDescription,
+				eventDescription: eventSchedule.description,
+				eventLocationName: eventSchedule.location?.name,
+				eventLocationLink: eventSchedule.location?.link,
```

Comment:
note: we should probably html escape this properly but it's already not handled correctly. Would need a separate issue to do a pass and see if it can be fixed at the source (the templating engine)